script.js

```diff
@@ -175,6 +175,36 @@ form.addEventListener('submit', (event) => {
   }
   if (emailValid) {
     form.submit();
+    form.reset();
+  } else {
+    showError();
   }
-  showError();
 });
+
+// FORM'S LOCAL STORAGE \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+const formElements = document.querySelectorAll('#contact-form *');
+const elementsArr = Array.from(formElements);
+// save the form's elements' values into the local storage
+function storeFormValues(element) {
+  localStorage.setItem(element.id, element.value);
+}
+// loop through all these elements
+for (let i = 0; i < elementsArr.length; i += 1) {
+  // set an onchange function that saves the value in the form's element
+  elementsArr[i].onchange = () => storeFormValues(elementsArr[i]);
+}
+
+//  REPOPULATE FORM'S ELEMENTS' CONTENT \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
+
+function getFromStorage(id) {
+  return localStorage.getItem(id);
+}
+
+// loads from local storage and updates the form's content
+function repopulateForm() {
+  for (let i = 0; i < elementsArr.length; i += 1) {
+    elementsArr[i].value = getFromStorage(elementsArr[i].id);
+  }
+}
+
+window.onload = repopulateForm();
```

Comment:
- [ ] Please ensure your contact form information is persistent to page reloads. After filling the form, and not submitting it, the entered values should be saved in the browser. Also when visiting the page the form inputs should be contained, and auto-populated with, the values entered in the previous visit to the page. Currently, your form is not preserving the form data in the localStorage of the browser so make sure that you fix this issue.
  
  [Screencast from 07-28-2023 07:58:26 PM.webm](https://github.com/Estete9/Fullstack-Portfolio/assets/92303891/0d3c043a-988c-471e-b390-503edda2bd41)

Reply:
Hey! I'm very grateful for your feedback. 
Can you show me how to reproduce the error? As you can see in this video, I did the same as in yours but in mine, the information was preserved.
https://www.loom.com/share/ec13d4aae94d4db18ad47e0c1eefb970